Lineage UI documentation (#544)
astronomer/docs: astro/lineage.md

To see details on the assertions that have failed, hover over a point on the chart.

### Distinct Count (Column-level)

The Distinct Count chart shows the total number of distinct values for a given column.

A distinct count can sometimes grow unexpectedly, perhaps if a successful campaign creates a set of new `customer_id` values in an order table suddenly. However, it can also suggest an underlying problem if, for example, a `menu_item_id` field shows that thousands of new menu items have been added overnight.

### Null (Column-level)

The Null chart shows the number of rows in the dataset where a given column contains a null value.

A large number of null values can be normal, such as when most orders on your system do not include a discount. However, an increase in null values on a column representing a ubiquitous piece of data, such as `quantity`, might indicate an issue.

### Quality Metrics (Column-level)

The Quality Metrics chart shows the pass/fail status of quality assertions in a Great Expectations suite for a given column.

To see details on the assertions that have failed, hover over a given point on the chart.

> ### Using the Compare Tab

Author: ```suggestion
### Compare Lineage Between Runs Using the Compare Tab
```

Author: Should we make this an `h2` even though it's part of the information pane?

---

> sidebar_label: 'Data Lineage'

Author: Hoping to put the overview in "Reference", so I think it's okay to keep this general. Hoping that this is the only lineage-related doc under "Observability"

> title: "Observe Deployments Through Data Lineage"

Author: Explore sounds a bit introductory, if we're going for utility I think we should be specific about what people can do with lineage (observe)

id: lineage
---
## Overview

> This guide explains how to navigate the Lineage graph and filter its data to better diagnose potential issues with your data pipelines.

Author: Cutting this down p significantly, I think that the overview sentence should cut to the chase. We can use some of this language in the following paragraphs.

For example, the following lineage graph illustrates the relations between parts of a restaurant order information pipeline as a series of jobs (nodes with a gear icon) and datasets (nodes with a database icon).

Relationships between your DAGs and datasets are represented by vertices, and potential errors for datasets and jobs are represented by the shape and color of a node.

![Lineage graph example](/img/docs/lineage-overview.png)

Using this visualization, it becomes clear how upstream errors can affect downstream jobs, as well as how downstream errors might have originated from upstream errors. Based on your integrations, the nodes of this graph can represent parts of your data pipeline both within and beyond your Astronomer Deployment.

Because the data lineage can map the full journey of your datasets, you can use this information to diagnose errors from the **Logs** and **Metrics** tabs. For example, if you receive an Airflow error log that a DAG failed to load data into an S3 bucket, you can go to the **Lineage** tab to see where your dataset became invalid, even if that occurred before your DAG started running.

> ## Navigating the Lineage Graph

Reviewer: I feel like we're using `Lineage Graph` and `Lineage tab` interchangeably, but we might need to take a step back here and have a small section that explains:

- Explore
- Issues
- Lineage

And then dive into the 3rd of course in the most detail. I have Conor feedback that there shouldn't be two tabs named "lineage" ha but for now we should just standardize on distinct naming

Reviewer: Honestly it might be good to take a general/wide screenshot or GIF of the lineage graph and like put boxes around the sections to show:

- Graph
- Node
- Information Pane

Kind of a challenge to put this view into words! Maybe we can do some research on how observability/visual tools document stuff?

Author: In the future we could toy with having more detailed screenshots, but I'm hesitant to compile too much information into an inaccessible format. For now, I took the overview screenshot and used it here, then described and example of a job/ dataset/ vertex within the context of the screenshot.

> In the lineage tab, Astronomer renders your data pipeline as a set of **dataset** and **job** nodes connected in a graph:

Reviewer: ```suggestion
In the lineage tab, Astronomer renders your data pipeline as a set of **dataset** and **job** nodes connected by directed vertices in a graph:
```

You introduce the word `directed vertices` below -- maybe just use it in context first to make it clear?

Author: We can use the phrase "directed graph" to describe this without having to get into the details of vertices, which I think is cleaner

> - Each job represents an individual step in your data pipeline, such as an Airflow task in your Deployment.
> - Each dataset represents a data source that your tasks interact with, such as a Snowflake database.

Reviewer: ```suggestion
- Each job represents an individual step in your data pipeline. This could be an Airflow task within a DAG or a Spark job in Databricks.
- Each dataset represents a data source that your tasks interact with. This could be a Snowflake database.
```

Stupid Q - Could dataset be something other than a database?

Stupid Q - the nodes here _could_ be

Author: "This could" makes it less clear whether the example is of a data source/ step or a dataset/job (grammatically, it should be the former). 

I don't think a dataset could be anything other than a database. The type of database might vary, but if you're managing data in any way you have to interact with a database/set

> Directed vertices connect jobs to datasets and vice versa. A single vertex will never connect two jobs or two datasets together.

Reviewer: ```suggestion
Directed vertices connect jobs to datasets and visa versa. A single vertex will never connect two jobs or two datasets to each other.
```

Author: never heard visa before, p sure it's vice

> You can navigate the graph by clicking and dragging your mouse across the screen. You can zoom in/out to a specific part of the graph by scrolling with your mouse wheel/ mousepad or by clicking the magnifying glass icons next on the information pane.

Reviewer: ```suggestion
To navigate the graph, click and drag your mouse across the screen or zoom in or out by scrolling. You can also click the magnifying glass icons in the information pane on the bottom-left of the screen.
```

To learn more information about a dataset or job, you can either hover over or click its node. Hovering over a node gives you high level information about the node at a glance. Specifically you'll see:

> - **Namespace**: Your Astronomer Namespace
> - **Name**: Your DAG ID and task ID, formatted as `<dag-id>.<task-id>`
> - **Run information (Job only)**: Metadata and status information about a job run
> - **Quality checks (Dataset only)**: The status of each completed data quality check for a dataset

Reviewer: ```suggestion
- **Namespace**: The namespace of the Deployment in which the job ran.
- **Name**: The DAG ID and task ID of the job, formatted as `<dag-id>.<task-id>`
- **Run information (job only)**: Metadata and status information about a job run
- **Quality checks (dataset only)**: The status of each completed data quality check for a dataset
```

What happens if this is a spark job -- is namespace empty?

Author: From what I understand, if the Spark job was triggered by a DAG in a namespace, then the job will also appear under that namespace grouping. Can you confirm @conorbev ?

Clicking a node populates the information pane with detailed information about the node. For more information about how to use this view, see [Using the Information Pane](lineage.md#using-the-information-page).

### Graph Legend

> Clicking on the key icon in the information pane opens the graph legend. The legend provides a visual guide to help you distinguish between:

Reviewer: ```suggestion
Clicking on the key icon in the information pane opens the graph legend. The legend provides a visual guide to help you distinguish between:
```

Does this key icon have a name?

Author: Nope, it definitely should because there's some important settings in there. One for PB?

- Job nodes and Dataset nodes.
- User-selected and unselected nodes.
- Completed, running, failed, and aborted jobs

> - Completed, running, failed, and aborted dataset checks

Reviewer: Are we capitalizing job and dataset or nah? Let's add to style guide

Author: Let's not capitalize them, and yeah I'll add that to style!

---
sidebar_label: 'Data Lineage'
title: "Observe Deployments Through Data Lineage"
id: data-lineage
---
## Overview

This guide explains how to navigate the Lineage graph and configure different views to explore the movement of your data and diagnose issues with your data pipelines.

> The **Lineage** tab is available to all members of your Astro Organization. It includes real-time metadata genealogy and analytics across Deployments. This information is rendered as a graph that spans your entire data pipeline infrastructure, including individual DAGs as well as data sources and database connections. Relationships between your DAGs and datasets are represented by vertices, and potential errors for datasets and jobs are represented by the shape and color of a node.

Reviewer: ```suggestion
The Lineage tab on Astro has three primary views:

- An **Explore** page that includes a real time view of all jobs that emit data lineage in your ecosystem. This includes metadata for Airflow tasks as well as jobs from other tools you may have integrated, such as Spark or dbt. This view also summarizes total job runs over time for up to one year.
- An **Issues** page that shows information about any job or dataset that may have caused an issue for a running pipeline.
- A **Lineage** page with a graph view that visualize data movements across DAGs, tasks, and Deployments. The graph primarily shows relationships between datasets and which pipelines and tasks actually connect them.

Generally speaking, these views can help you map your organization's data ecosystem and diagnose errors that may otherwise be difficult to identify across environments and tools. For example, if an Airflow task failed because it was unable to load data into an S3 bucket, you might go to the **Lineage** tab on Astro to determine the source of the problematic data and the point at which it became invalid.

Views in the **Lineage** tab are available to all members of your Organization on Astro.
```

> Because data lineage maps the full journey of your datasets, you can use this information to diagnose errors you detect from your Deployment logs and metrics. For example, if there was an error in an Airflow task log explaining that a task failed to load data into an S3 bucket, you could go to the **Lineage** tab on Astro to see the source of the problematic data and the point at which point it became invalid.

Reviewer: ```suggestion
```

## Lineage Tab Overview

The Lineage tab is composed of three different pages available in a left-hand sidebar:

> - **Explore:** Summarizes the performance of all data pipeline runs over the last year and lists your most recent pipeline runs

Reviewer 2: Add: `. Click on any run to view its Lineage graph`

> - **Issues:** Shows information about each job or dataset that caused an issue during a pipeline run

Reviewer 2: Add: `. Click on any job or dataset issue to view its Lineage graph.`

> - **Lineage:** By default, renders a data lineage graph for your most recent pipeline run

Reviewer 2: `By default, renders a data lineage graph for your most recent pipeline run` --> `By default, renders your most recently viewed graph. First click on a job or dataset of interest from the Explore or Issues pages if no default has been set yet.`
